mlflow/models/docker_utils.py

```diff
@@ -37,6 +38,32 @@
 RUN pip install virtualenv
 """
 
+if os.getenv("http_proxy") is not None and os.getenv("https_proxy") is not None:
+
+    # Expects proxies as PROTOCOL://HOSTNAME:PORT
+    parsed_http_proxy = urlparse(os.environ["http_proxy"])
+    assert parsed_http_proxy.hostname is not None, "Invalid `http_proxy` hostname."
+    assert isinstance(parsed_http_proxy.port, int), f"Invalid Proxy Port: {parsed_http_proxy.port}"
+
+    parsed_https_proxy = urlparse(os.environ["https_proxy"])
+    assert parsed_https_proxy.hostname is not None, "Invalid `https_proxy` hostname."
+    assert isinstance(
+        parsed_https_proxy.port, int
+    ), f"Invalid Proxy Port: {parsed_https_proxy.port}"
+
+    MAVEN_PROXY = (
+        " -DproxySet=true -Dhttp.proxyHost={http_proxy_host} "
+        "-Dhttp.proxyPort={http_proxy_port} -Dhttps.proxyHost={https_proxy_host} "
+        "-Dhttps.proxyPort={https_proxy_port} -Dhttps.nonProxyHosts=repo.maven.apache.org"
```

Comment:
If the http(s)_proxy contains username/password, shall set them in maven proxy config ?

Reply:
I'm not that well versed with `maven`, but I believe we can pass username and password as 

` MAVEN_PROXY += " -Dhttp.proxyUser={proxy_username} -Dhttp.proxyPassword={proxy_password}"`

Could you please confirm, thanks!

Reply:
I don't have env at hand, but I think this is correct. 
-Dhttp.proxyUser={proxy_username} -Dhttp.proxyPassword={proxy_password}

Ref: https://www.baeldung.com/maven-behind-proxy

Reply:
Implemented the changes in the latest PR commit and have also modified the unit tests to validate authenticated proxy servers.